### DISCUSSION TITLE
How to customize errors throwed from static queries?

### DISCUSSION THREAD

**Discussion by tadeas22** [gatsbyjs/gatsby#30105] [Help]
Hello

is it possible to custom errors throwed from static queries? For instance i can customize schema to require field `foo` in `siteMetadata` .

```javascript
exports.createSchemaCustomization = ({ actions }) => {
	const { createTypes } = actions;
	const typeDefs = `
          type SiteSiteMetadata implements Node {
            foo: String!
          }
        `;
	createTypes(typeDefs);
};
```

`src/pages/index.js`

```javascript
export const query = graphql`
	{
		site {
			siteMetadata {
				foo
			}
		}
	}
`;
```

and i need throw custom error/exception when missing

now i am getting the following error and i want customize text to "Foo is required because it contains secret" for instance
```
...
error There was an error in your GraphQL query:

Cannot return null for non-nullable field SiteSiteMetadata.foo.

The field "SiteSiteMetadata.foo." was expli...
```

**Comment by LekoArts** [comment]
Hi, thanks for the question!

No, this isn't currently possible. Depending on how the information for `foo` comes in you could validate it's existence in the `onPreBootstrap` phase (you have access to plugin options there) or in the `gatsby-config.js` file itself (you can throw errors there).

If you're creating a plugin: https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/configuring-usage-with-plugin-options/
